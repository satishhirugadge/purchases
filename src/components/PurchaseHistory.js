import React from "react";
import { usePurchaseDetails } from "../context/usePurchaseDetails";

export const PurchaseHistory = () => {
  const { purchaseData } = usePurchaseDetails();
  return (
    <div className="purchase-table">
      {purchaseData.length > 0 && (
        <table>
          <thead>
            <tr>
              <th>Transaction Id</th>
              <th>Date</th>
              <th>Amount</th>
              <th>Total rewards</th>
            </tr>
          </thead>
          <tbody>
            {purchaseData.map(({ id, date, amount, rewards }) => (
              <tr key={id}>
                <td>{id}</td>
                <td>{date}</td>
                <td>{amount}</td>
                <td>{rewards}</td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
};
