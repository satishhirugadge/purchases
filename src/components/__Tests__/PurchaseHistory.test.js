import React from "react";
import { render, screen } from "@testing-library/react";
import { PurchaseHistory } from "../PurchaseHistory";
import { PurchaseContext } from "../../providers/PurchaseDataProvicer";
import { purchasesMockData } from "./MockData";

test("verifying table row data ", () => {
  render(
    <PurchaseContext.Provider value={{ purchaseData: purchasesMockData }}>
      <PurchaseHistory />
    </PurchaseContext.Provider>
  );

  expect(screen.getByText("11/10/2021")).toBeInTheDocument();
});

test("snapshot test", () => {
  const component = render(
    <PurchaseContext.Provider value={{ purchaseData: purchasesMockData }}>
      <PurchaseHistory />
    </PurchaseContext.Provider>
  );
  expect(component).toMatchSnapshot();
});
