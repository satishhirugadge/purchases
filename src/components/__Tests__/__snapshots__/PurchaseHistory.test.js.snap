// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="purchase-table"
      >
        <table>
          <thead>
            <tr>
              <th>
                Transaction Id
              </th>
              <th>
                Date
              </th>
              <th>
                Amount
              </th>
              <th>
                Total rewards
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                201
              </td>
              <td>
                01/10/2021
              </td>
              <td>
                120
              </td>
              <td />
            </tr>
            <tr>
              <td>
                202
              </td>
              <td>
                11/10/2021
              </td>
              <td>
                450
              </td>
              <td />
            </tr>
            <tr>
              <td>
                203
              </td>
              <td>
                01/10/2021
              </td>
              <td>
                620
              </td>
              <td />
            </tr>
            <tr>
              <td>
                204
              </td>
              <td>
                12/10/2021
              </td>
              <td>
                420
              </td>
              <td />
            </tr>
            <tr>
              <td>
                205
              </td>
              <td>
                15/9/2021
              </td>
              <td>
                250
              </td>
              <td />
            </tr>
            <tr>
              <td>
                206
              </td>
              <td>
                01/6/2021
              </td>
              <td>
                860
              </td>
              <td />
            </tr>
            <tr>
              <td>
                207
              </td>
              <td>
                01/9/2021
              </td>
              <td>
                990
              </td>
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="purchase-table"
    >
      <table>
        <thead>
          <tr>
            <th>
              Transaction Id
            </th>
            <th>
              Date
            </th>
            <th>
              Amount
            </th>
            <th>
              Total rewards
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              201
            </td>
            <td>
              01/10/2021
            </td>
            <td>
              120
            </td>
            <td />
          </tr>
          <tr>
            <td>
              202
            </td>
            <td>
              11/10/2021
            </td>
            <td>
              450
            </td>
            <td />
          </tr>
          <tr>
            <td>
              203
            </td>
            <td>
              01/10/2021
            </td>
            <td>
              620
            </td>
            <td />
          </tr>
          <tr>
            <td>
              204
            </td>
            <td>
              12/10/2021
            </td>
            <td>
              420
            </td>
            <td />
          </tr>
          <tr>
            <td>
              205
            </td>
            <td>
              15/9/2021
            </td>
            <td>
              250
            </td>
            <td />
          </tr>
          <tr>
            <td>
              206
            </td>
            <td>
              01/6/2021
            </td>
            <td>
              860
            </td>
            <td />
          </tr>
          <tr>
            <td>
              207
            </td>
            <td>
              01/9/2021
            </td>
            <td>
              990
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
